pkg/osquery/extension.go

```diff
@@ -507,6 +529,10 @@ func (e *Extension) GenerateConfigs(ctx context.Context) (map[string]string, err
 		confBytes, _ = e.knapsack.ConfigStore().Get([]byte(configKey))
 
 		if len(confBytes) == 0 {
+			if !e.enrolled() {
+				// Not enrolled yet -- return an empty config
+				return map[string]string{"config": "{}"}, nil
+			}
```

Comment:
I'm not positive we even need this? If I omit it, we get:

```
{
"time":"2024-02-15T18:55:35.676293Z",
"level":"INFO",
"source":{
  "function":"github.com/kolide/launcher/ee/log/osquerylogs.(*OsqueryLogAdapter).Write",
  "file":"/Users/rebeccamahany-horton/Repos/launcher/ee/log/osquerylogs/log.go","line":83},
  "msg":"W0215 13:55:35.676178 -347473728 init.cpp:760] Error reading config: error getting config: loading config failed, no cached config: enrollment invalid, reenrollment errored: could not read enroll secret: could not read enroll secret path /etc/kolide-k2/secret: open /etc/kolide-k2/secret: no such file or directory",
  "launcher_run_id":"01HPQ1ZGS0R3Q0WAZ7FDB4N7FV",
  "component":"osquery",
  "osqlevel":"stderr",
  "caller":"init.cpp:760"
}
```

And behavior doesn't otherwise change -- as soon as I make the secret file available, launcher completes enrollment and queries start to run.